### PR TITLE
Refactor playground mocks folder structure

### PR DESCRIPTION
**Pull request by haishengwu-okta**
## Description:

after refactor

```
playground/mocks/
  - config                       <-- helpers 
  - data                          <-- JSON mocks
  - spec-okta-api.        <-- dyson API mock setup for authn and idx
  - spec-device-authenticator     <- dyson API mock setup for external device authenticator
```

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-303165](https://oktainc.atlassian.net/browse/OKTA-303165)

### CODE REVIEW

**Comment by magizh-okta** [comment]
@haishengwu-okta can u add some info on how this helps to get rid of the method error ?

**Comment by haishengwu-okta** [comment]
@magizh-okta 
mock data and spec are living in same directory where dyson recursively search all files and look for `method` recursively of the object return by any file.
the refactor separates mock data (json) with specs.

**Comment by pradeepdewda-okta** [comment]
@haishengwu-okta i think add a ReadMe also in this commit which briefly describe how mock data is used. Later on we can keep improving it.

**Comment by pradeepdewda-okta** [comment]
> @haishengwu-okta i think add a ReadMe also in this commit which briefly describe how mock data is used. Later on we can keep improving it.

for eg. the info about the little flip we do `${xs[0]}`

File: test/testcafe/spec/Success_spec.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import IdentityPageObject from '../framework/page-objects/IdentityPageObject';
 import SuccessPageObject from '../framework/page-objects/SuccessPageObject';
 import { RequestMock } from 'testcafe';
-import success from '../../../playground/mocks/idp/idx/data/success';
-import identify from '../../../playground/mocks/idp/idx/data/identify';
+import success from '../../../playground/mocks/data/idp/idx//success';
```
Comment (anipendakur-okta) [inline test/testcafe/spec/Success_spec.js:4]: there's a double `//`

File: test/testcafe/spec/Success_spec.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+import identify from '../../../playground/mocks/data/idp/idx//identify';
```
Comment (anipendakur-okta) [inline test/testcafe/spec/Success_spec.js:5]: there's a double `//`

File: test/testcafe/spec/TerminalReturnEmail_spec.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import TerminalPageObject from '../framework/page-objects/TerminalPageObject';
 import { RequestMock } from 'testcafe';
-import terminalReturnEmail from '../../../playground/mocks/idp/idx/data/terminal-return-email';
-import terminalTransferedEmail from '../../../playground/mocks/idp/idx/data/terminal-transfered-email';
+import terminalReturnEmail from '../../../playground/mocks/data/idp/idx//terminal-return-email';
+import terminalTransferedEmail from '../../../playground/mocks/data/idp/idx//terminal-transfered-email';
```
Comment (anipendakur-okta) [inline test/testcafe/spec/TerminalReturnEmail_spec.js:4]: there's a double `//`

File: test/testcafe/spec/Success_spec.js (hunk 1 above)
Comment (pradeepdewda-okta) [reply test/testcafe/spec/Success_spec.js:4]: good catch..

**Pull request merged**